sdk/inc/azure/core/az_mqtt5_property_bag.h

```diff
@@ -148,6 +103,13 @@ AZ_NODISCARD az_mqtt5_property_bag_options az_mqtt5_property_bag_options_default
  */
 AZ_NODISCARD az_result az_mqtt5_property_bag_empty(az_mqtt5_property_bag* property_bag);
 
+/**
+ * @brief Destroys a MQTT 5 property bag instance.
+ *
+ * @param[in] property_bag A pointer to an #az_mqtt5_property_bag instance to destroy.
+ */
+void az_mqtt5_property_bag_destroy(az_mqtt5_property_bag* property_bag);
```

Comment:
I don't think we should have this: our SDK should not be able to allocate or de-allocate resources.